src/Autoload/ScoperAutoloadGenerator.php

```diff
@@ -103,7 +103,7 @@ public function dump(): string
                 namespace {
                     \$loader = (static function () {
                         // Backup the autoloaded Composer files
-                        \$existingComposerAutoloadFiles = \$GLOBALS['__composer_autoload_files'] ?? [];
+                        \$existingComposerAutoloadFiles = isset(\$GLOBALS['__composer_autoload_files']) ? \$GLOBALS['__composer_autoload_files'] : [];
```

Comment:
wouldn't the `trait_exists()` call L47 fail too?

Reply:
`trait_exists` is present on PHP >= 5.4
https://www.php.net/manual/en/function.trait-exists.php

But yeah, an automated check of the generated file would be great, I only encountered this while trying to set up PHP 5.6 compatibility testing, which was a royal pain itself... I don't really know hoe to do that, maybe run it through PHPCS with PHPCompatibility checks?

Reply:
Guess the easiest would be via a docker image, doing an e2e test like there is in the `.makefile/e2e`. But hardly a requirement